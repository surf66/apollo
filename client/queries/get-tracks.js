import gql from "graphql-tag";

export const GET_TRACKS = gql`
  {
    tracks {
      title
      artist
    }
  }
`;