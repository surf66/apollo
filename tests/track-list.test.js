import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import TrackList from '../client/components/track-list';
import { GET_TRACKS } from '../client/queries/get-tracks';

const mockGraphQlResponse = [
  {
    request: {
      query: GET_TRACKS
    },
    result: {
      data: {
        tracks: [
          {
            title: "Every Day",
            artist: "Eric Prydz"
          },
          {
            title: "Opus",
            artist: "Eric Prydz"
          }
        ]
      }
    }
  }
];

test('should match snapshot when tracks provided', async () => {
  const wrapper = mount(
    <MockedProvider mocks={mockGraphQlResponse} addTypename={false}>
      <TrackList />
    </MockedProvider>
  );

  await new Promise(resolve => setTimeout(resolve));
  wrapper.update();

  expect(wrapper.html()).toMatchSnapshot();
});

test('should match snapshot when waiting for data', async () => {
  const wrapper = shallow(
    <MockedProvider mocks={mockGraphQlResponse} addTypename={false}>
      <TrackList />
    </MockedProvider>
  );

  expect(wrapper.html()).toMatchSnapshot();
});