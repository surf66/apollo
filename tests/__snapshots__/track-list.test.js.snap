// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when tracks provided 1`] = `"<p><strong>Every Day</strong>, Eric Prydz</p>"`;

exports[`should match snapshot when waiting for data 1`] = `"Loading..."`;
